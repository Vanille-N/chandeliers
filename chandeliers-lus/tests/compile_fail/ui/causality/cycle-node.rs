chandeliers_lus::decl! {
    node foo(i : int) returns (o : int);
    let o = bar(i); tel;

    node bar(i : int) returns (o : int);
    let o = foo(i); tel;
}

fn main() {}
