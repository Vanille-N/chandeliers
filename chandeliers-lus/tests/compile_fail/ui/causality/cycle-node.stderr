error: Node `bar` was found to be part of a dependency cycle
 --> tests/fail/ui/causality/cycle-node.rs:5:10
  |
5 |     node bar(i : int) returns (o : int);
  |          ^^^
  |
note: The cycle also goes through Node `foo`
 --> tests/fail/ui/causality/cycle-node.rs:2:10
  |
2 |     node foo(i : int) returns (o : int);
  |          ^^^
